contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,395 @@
+package mgo
+
+import (
+	"context"
+
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type mongoConfig struct {
+	serviceName string
+}
+
+// MongoOption represents an option that can be passed to Dial
+type MongoOption func(*mongoConfig)
+
+func defaults(cfg *mongoConfig) {
+	cfg.serviceName = "mongodb"
+}
+
+// WithServiceName sets the service name for a given MongoDB context.
+func WithServiceName(name string) MongoOption {
+	return func(cfg *mongoConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(ctx context.Context, url string, opts ...MongoOption) (*WrapSession, error) {
+	session, err := mgo.Dial(url)
+	s := &WrapSession{
+		Session: session,
+		ctx:     ctx,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// WrapSession wraps a Session with the Context and Config information
+// needed to support tracing.
+type WrapSession struct {
+	*mgo.Session
+
+	ctx context.Context
+	cfg mongoConfig
+}
+
+func newChildSpanFromContext(ctx context.Context, config mongoConfig, resource string, op string) ddtrace.Span {
+	name := op
```

Comment:
it looks like `op` is always `mongodb.query`. Can we remove it as an argument?

Reply:
(sorry, `resource`)

Based on feedback from https://github.com/DataDog/dd-trace-go/pull/294/files#r207120793 I think both need to be set to `mongodb.query`

Reply:
This may require further discussion, when I was testing, just having "mongodb.query" didn't give a lot of insight into what each span was doing on the Trace webpage.

Reply:
Yeah, it sounds like its necessary for the scrubbing to work properly:

> Although the tag name you've chosen is the correct standard used in most tracers, we have to use mongodb.query. Our choice of tag naming was rather uneven, you can see the span types and tag names we use in the agent.

```go
	case "mongodb":
		o.obfuscateJSON(span, "mongodb.query", o.mongo)
```

Reply:
Got it.  I'll change everything to `mongodb.query`.

Reply:
Done.